### PR TITLE
[Week 5] Improve Honor assignment

### PR DESCRIPTION
**Pull request by dniku**
This PR fixes _many_ problems with the Week 5 Honor assignment. In no particular order:

* Fully support Colab. In fact, I authored this notebook in Colab
* Replace `skimage.transform.resize` with `cv2.resize` and achieve a 6x evaluation / 2x training speedup.
* Refactor `atari_util.py`: cleanup code, support `dim_order='pytorch'` (functionally equivalent to `'theano'` though)
* Embed neural net architecture into the notebook, to avoid problems in case postimg goes down or changes domain again
* Fix title image to point to Wikipedia, which is unlikely to go down
* Only perform reward scaling in the training loop itself
* Start training loop from `len(entropy_history)` (a cheap way to support resuming training)
* Terminate training after exceeding a threshold that is computed from unscaled reward
* Fix crash due to `% color`
* Some other general cleanup
* Only perform evaluation 10 times instead of 100 because it takes ~10 seconds in Colab to evaluate 1 episode

### CODE REVIEW

File: week5_policy_based/practice_a3c.ipynb [hunk 1]
```diff
@@ -455,29 +493,37 @@
...
    "outputs": [],
    "source": [
-    "for i in trange(100000):\n",
-    "    batch_actions = agent.sample_actions(agent.step(batch_states))\n",
+    "for i in trange(len(entropy_history), 100000):\n",
+    "    agent_outputs = agent.step(batch_states)\n",
+    "    batch_actions = agent.sample_actions(agent_outputs)\n",
     "    batch_next_states, batch_rewards, batch_done, _ = env_batch.step(batch_actions)\n",
     "\n",
+    "    # We scale rewards to avoid exploding gradients during optimization.\n",
```
Comment (yhn112) [inline week5_policy_based/practice_a3c.ipynb:501]: Not sure if it is really exploding

Comment (dniku) [reply week5_policy_based/practice_a3c.ipynb]: Rewrote explanation
